platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/webview/WebViewCapture.kt

```diff
@@ -0,0 +1,258 @@
+            ERROR_UNSUPPORTED_SCHEME      -> "UNSUPPORTED_SCHEME"
+            ERROR_FAILED_SSL_HANDSHAKE    -> "FAILED_SSL_HANDSHAKE"
+            ERROR_BAD_URL                 -> "BAD_URL"
+            ERROR_FILE                    -> "FILE"
+            ERROR_FILE_NOT_FOUND          -> "FILE_NOT_FOUND"
+            ERROR_TOO_MANY_REQUESTS       -> "TOO_MANY_REQUESTS"
+            ERROR_UNSAFE_RESOURCE         -> "UNSAFE_RESOURCE"
+            else                          -> "UNKNOWN"
+        }
+
+    private data class PageLoadRequest(
+        val url: String,
+        val isError: Boolean = false,
+        val errorCode: String? = null,
+        val errorCodeName: String? = null,
+        val errorDescription: String? = null,
+    ) {
+        val fields: Map<String, String> by lazy {
+            buildMap {
+                put("_url", url)
+                if (isError) {
+                    errorCode?.let { put("_errorCode", it) }
+                    errorCodeName?.let { put("_errorCodeName", it) }
+                    errorDescription?.let { put("_description", it) }
+                }
+            }
+        }
+    }
+    /**
+     * Companion object for WebViewCapture.
+     */
+    companion object {
+        private const val TAG = "WebViewCapture"
+        private const val BRIDGE_NAME = "BitdriftLogger"
+
+        /**
+         * Instruments a WebView to capture page load events, Core Web Vitals,
+         * and network requests.
+         *
+         * This method:
+         * - Wraps the existing WebViewClient to capture page load events
+         * - Enables JavaScript execution
+         * - Injects the Bitdrift JavaScript bridge at document start
+         * - Registers a native interface for receiving bridge messages
+         *
+         * @param webview The WebView to instrument
+         * @param logger Optional logger instance. If null, uses Capture.logger()
+         */
+        @SuppressLint("SetJavaScriptEnabled", "RequiresFeature")
+        @JvmStatic
+        @JvmOverloads
+        fun instrument(webview: WebView, logger: ILogger? = null) {
+            val effectiveLogger = logger ?: Capture.logger()
+            
+            // Wrap existing WebViewClient
+            val capture = if (WebViewFeature.isFeatureSupported(WebViewFeature.GET_WEB_VIEW_CLIENT)) {
+                val original = WebViewCompat.getWebViewClient(webview)
+                WebViewCapture(original, effectiveLogger)
+            } else {
+                WebViewCapture(WebViewClient(), effectiveLogger)
+            }
+            
+            webview.webViewClient = capture
+
+            // Enable JavaScript (required for bridge)
+            webview.settings.javaScriptEnabled = true
+
+            // Register JavaScript interface for receiving bridge messages
+            val bridgeHandler = WebViewBridgeHandler(effectiveLogger, capture)
+            webview.addJavascriptInterface(bridgeHandler, BRIDGE_NAME)
```

Comment:
if this works well we could just use this approach for everything and replace the current `webview.pageLoad` span and forego the need for `androidx.webkit.WebViewCompat` altogether